src/IB/IBStandardForceGen.cpp

```diff
@@ -228,56 +228,44 @@ IBStandardForceGen::initializeLevelData(const Pointer<PatchHierarchy<NDIM> > hie
-                   d_target_point_data[level_number].petsc_node_idxs.begin(),
-                   std::bind2nd(std::multiplies<int>(), NDIM));
-    std::transform(d_target_point_data[level_number].petsc_global_node_idxs.begin(),
-                   d_target_point_data[level_number].petsc_global_node_idxs.end(),
-                   d_target_point_data[level_number].petsc_global_node_idxs.begin(),
-                   std::bind2nd(std::multiplies<int>(), NDIM));
+    std::for_each(d_spring_data[level_number].petsc_mastr_node_idxs.begin(),
+                  d_spring_data[level_number].petsc_mastr_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_spring_data[level_number].petsc_slave_node_idxs.begin(),
+                  d_spring_data[level_number].petsc_slave_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_spring_data[level_number].petsc_global_mastr_node_idxs.begin(),
+                  d_spring_data[level_number].petsc_global_mastr_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_spring_data[level_number].petsc_global_slave_node_idxs.begin(),
+                  d_spring_data[level_number].petsc_global_slave_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+
+    std::for_each(d_beam_data[level_number].petsc_mastr_node_idxs.begin(),
+                  d_beam_data[level_number].petsc_mastr_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_beam_data[level_number].petsc_next_node_idxs.begin(),
+                  d_beam_data[level_number].petsc_next_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_beam_data[level_number].petsc_prev_node_idxs.begin(),
+                  d_beam_data[level_number].petsc_prev_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_beam_data[level_number].petsc_global_mastr_node_idxs.begin(),
+                  d_beam_data[level_number].petsc_global_mastr_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_beam_data[level_number].petsc_global_next_node_idxs.begin(),
+                  d_beam_data[level_number].petsc_global_next_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_beam_data[level_number].petsc_global_prev_node_idxs.begin(),
+                  d_beam_data[level_number].petsc_global_prev_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+
+    std::for_each(d_target_point_data[level_number].petsc_node_idxs.begin(),
+                  d_target_point_data[level_number].petsc_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
+    std::for_each(d_target_point_data[level_number].petsc_global_node_idxs.begin(),
+                  d_target_point_data[level_number].petsc_global_node_idxs.end(),
+                  [](int& i) { i *= NDIM; });
```

Comment:
Could we rewrite this with an outer loop over vectors and then a range-for over their entries?